### ISSUE TITLE
Deserialization order issue

### ISSUE THREAD

**Issue by TenorD1** [DigDes/SoapCore#733]
I'am trying to use SoapCore library in order to replace WCF SOAP service, but i have deserailization issue because the fields order did not respect alphabetical order. It seems the problem comes from xml serialisation (known on Microsoft platform). I use DataContract attribute for my DTO object and add attribute  ServiceContract with XmlSerializerFormat option without success.
I am not an expert on both WCF and SOAP maybe simple solution, some help are welcome.

**Comment by andersjonsson** [comment]
I think you can force a certain order by adding `[XmlElement]` with `Order` to the members of your DTO.
Don't know if that will work for your purposes though.

`[DataMember]` also has an `Order` property but I'm guessing that `XmlSerializer` is the one that's used here

```
class DTO
{
    [XmlElement(Order = 2)]
    public string Banana {get;set;}

    [XmlElement(Order = 1)]
    public string Apple {get;set}
}
```

**Comment by TenorD1** [comment]
Thanks for the help, and the proposal solution given.
As usual I found a solution between i send this issue and your response.
In fact, I focus on MessageBodyMember attribute and option order that did not work, but DataContract attribute as you mentioned has an option Order too, that do the job and that i missed.
Unfortunately, I have another issue to offer the same service without change in order to avoid introducing breaking change for our client.
In fact, i think the problem is the same as an issue closed without solution  #550 Soap Body IsWrapped for requests - DataContractSerializer vs XmlSerializer which reference another issue without solution too.
The problem is the soap structure from WCF service discard method and ppropose directly parameter structure whereas soapcore add this method before defining parameter structure.
